anta/custom_types.py

```diff
@@ -17,13 +18,40 @@ def aaa_group_prefix(v: str) -> str:
     return f"group {v}" if v not in built_in_methods and not v.startswith("group ") else v
 
 
+def interface_autocomplete(v: str) -> str:
+    """Allow the user to only provide the beginning of an interface name
+    E.g. et, eth will be changed to Ethernet"""
+    intf_id_re = re.compile(r"[0-9]+(\/[0-9]+)*(\.[0-9]+)?")
+    m = intf_id_re.search(v)
+    if m is None:
+        raise ValueError(f"Could not parse interface ID in interface '{v}'")
+    intf_id = m[0]
+    if any(v.lower().startswith(p) for p in ["et", "eth"]):
+        return f"Ethernet{intf_id}"
+    return v
+
+
+def interface_case_sensitivity(v: str) -> str:
+    """Reformat interface name to match expected case sensitivity.
+    E.g. ethernet -> Ethernet
+         vlan -> Vlan"""
+    if isinstance(v, str) and len(v) > 0 and not v[0].isupper():
+        return f"{v[0].upper()}{v[1:]}"
+    return v
+
+
 # ANTA framework
 TestStatus = Literal["unset", "success", "failure", "error", "skipped"]
 
 # AntaTest.Input types
 AAAAuthMethod = Annotated[str, AfterValidator(aaa_group_prefix)]
 Vlan = Annotated[int, Field(ge=0, le=4094)]
 Vni = Annotated[int, Field(ge=1, le=16777215)]
-Interface = Annotated[str, Field(pattern=r"^(Ethernet|Fabric|Loopback|Management|Port-Channel|Tunnel|Vlan|Vxlan)[0-9]+(\/[0-9]+)*(\.[0-9]+)?$")]
+Interface = Annotated[
+    str,
+    Field(pattern=r"^(Ethernet|Fabric|Loopback|Management|Port-Channel|Tunnel|Vlan|Vxlan)[0-9]+(\/[0-9]+)*(\.[0-9]+)?$"),
```

Comment:
Let's get prepared for WAN ;) 
```suggestion
    Field(pattern=r"^(Dps|Ethernet|Fabric|Loopback|Management|Port-Channel|Tunnel|Vlan|Vxlan)[0-9]+(\/[0-9]+)*(\.[0-9]+)?$"),
```